scheduler/task.go

```diff
@@ -231,6 +261,94 @@ func (t *task) Spin() {
 	}
 }
 
+// Fork stream stuff here
+func (t *task) stream() {
+	var consecutiveFailures int
+	for {
+		metricsChan, errChan, err := t.metricsManager.StreamMetrics(
+			t.id,
+			t.workflow.tags,
+			t.maxCollectDuration,
+			t.maxMetricsBuffer)
+		if err != nil {
+			consecutiveFailures++
+			e := checkTaskFailures(t, consecutiveFailures)
+			if e != nil {
+				return
+			}
+			// If we are unsuccessful at setting up the stream
+			// wait for a second and then try again until either
+			// the connection is successful or we pass the
+			// acceptable number of consecutive failures
+			time.Sleep(time.Second)
```

Comment:
Consider setting a local variable for the amount of time we will wait and use it here and below on line 318.  It caught my attention that the amounts were different.